openedx/core/djangoapps/programs/utils.py

```diff
@@ -461,57 +462,70 @@ def _attach_course_run_upgrade_url(self, run_mode):
     def _attach_course_run_may_certify(self, run_mode):
         run_mode['may_certify'] = self.course_overview.may_certify()
 
-    def _check_enrollment_for_user(self, course_run):
-        applicable_seat_types = self.data['applicable_seat_types']
-
-        (enrollment_mode, active) = CourseEnrollment.enrollment_mode_for_user(
-            self.user,
-            CourseKey.from_string(course_run['key'])
+    def _filter_out_courses_with_entitlements(self, courses):
```

Comment:
This method and `_filter_out_courses_with_enrollments` could benefit from a docstring

Reply:
should `_filter_out_courses_with_entitlements` also have a docstring?

Reply:
Yup.

Reply:
Include docstring for the arguments and return variables. Python is not strongly type, so it's useful to see what the argument we are passing in actually is.